lib/redis_memo/memoize_method.rb

```diff
@@ -64,144 +65,150 @@ def memoize_method(method_name, method_id: nil, **options, &depends_on)
     end
   end
 
-  def self.method_id(ref, method_name)
-    is_class_method = ref.class == Class
-    class_name = is_class_method ? ref.name : ref.class.name
+  class << self
```

Comment:
What's the motivation for moving these methods to be defined in the eigenclass instead?

Reply:
ah, this way I can use `private` to mark them as private class methods.

alternatively we can use the `private_class_method` keyword